js/dcf-tabs.js

```diff
@@ -0,0 +1,124 @@
+class DCFTabs {
+  constructor(tabGroups) {
+    this.tabGroups = tabGroups;
+  }
+
+  initialize() {
+    Array.prototype.forEach.call(this.tabGroups, (tabGroup) => {
+      // Define constants for tabs
+      const tabList = tabGroup.querySelector('.dcf-tabs > ol, .dcf-tabs > ul');
+      const tabs = tabList.querySelectorAll('a');
+      const panels = tabGroup.querySelectorAll('.dcf-tabs > div, .dcf-tabs > section');
+      const uuid = DCFUtility.uuidv4();
+
+      // Prefix each tab group with a unique ID.
+      tabGroup.setAttribute('id', uuid.concat('-tab-group'));
+
+      // Tab switching function
+      const switchTab = (oldTab, newTab) => {
+        newTab.focus();
+        // Make the active tab focusable by the user (Tab key)
+        newTab.removeAttribute('tabindex');
+        // Set the selected state
+        newTab.setAttribute('aria-selected', 'true');
+        oldTab.removeAttribute('aria-selected');
+        oldTab.setAttribute('tabindex', '-1');
+        // Get the indices of the new and old tabs to find the correct
+        // tab panels to show and hide
+        let index = Array.prototype.indexOf.call(tabs, newTab);
+        let oldIndex = Array.prototype.indexOf.call(tabs, oldTab);
+        panels[oldIndex].hidden = true;
+        panels[index].hidden = false;
+      };
+
+      // Tab styling and functions.
+      Array.prototype.forEach.call(tabs, (tab, tabIndex) => {
+        // Add class to each tab
+        tab.classList.add('dcf-tab', 'dcf-d-block');
+        // Prefix each tab within its parent tab group with the corresponding uuid.
+        let nextTab = tabIndex + DCFUtility.magicNumbers('int1');
+        tab.setAttribute('id', uuid.concat('-tab-', nextTab));
+        // Add role to each tab
+        tab.setAttribute('role', 'tab');
+        // Add tabindex to each tab
+        tab.setAttribute('tabindex', '-1');
+        // Add class to each tab's parent (list item)
+        tab.parentNode.classList.add('dcf-tabs-list-item', 'dcf-mb-0');
+        // Add role to each tab's parent (list item)
+        tab.parentNode.setAttribute('role', 'presentation');
+        // Handle clicking of tabs for mouse users
+        tab.addEventListener('click', (clickEvent) => {
+          clickEvent.preventDefault();
+          let currentTab = tabList.querySelector('[aria-selected]');
+          if (clickEvent.currentTarget !== currentTab) {
+            switchTab(currentTab, clickEvent.currentTarget);
+          }
+        });
+        // Handle keydown events for keyboard users
+        tab.addEventListener('keydown', (keydownEvent) => {
+          // Get the index of the current tab in the tabs node list
+          let index = Array.prototype.indexOf.call(tabs, keydownEvent.currentTarget);
+          // Work out which key the user is pressing and
+          // Calculate the new tab's index where appropriate
+          const int37 = 37;
+          const int39 = 39;
+          const int40 = 40;
```

Comment:
Add to DCFUtility magicNumbers object as `arrowDownCode` so it can be reused.  Also add one for `arrowUpCode` with value of `38` so all are covered.